src/jit/lsra.cpp

```diff
@@ -137,6 +158,64 @@ regMaskTP LinearScan::allRegs(RegisterType rt)
         return availableIntRegs;
 }
 
+//--------------------------------------------------------------------------
+// allRegs: represents a set of registers that can be used to allocate a
+// multi-reg call node.
+//
+// Arguments:
+//    call   -  Multi-reg call node
+//
+// Return Value:
+//    Mask representing the set of available registers for multi-reg call 
+//    node.
+//
+// Note:
+// Multi-reg call node available regs = Bitwise-OR(allregs(GetReturnRegType(i)))
+// for all i=0..RetRegCount-1.
+regMaskTP LinearScan::allMultiRegCallNodeRegs(GenTreeCall* call)
+{
+    assert(call->HasMultiRegRetVal());
+
+    ReturnTypeDesc* retTypeDesc = call->GetReturnTypeDesc();
+    regMaskTP resultMask = allRegs(retTypeDesc->GetReturnRegType(0));
+
+    unsigned count = retTypeDesc->GetReturnRegCount();
+    for (unsigned i = 1; i < count; ++i)
+    {
+        resultMask |= allRegs(retTypeDesc->GetReturnRegType(i));
+    }
+
+    return resultMask;
+}
+
+//--------------------------------------------------------------------------
+// allRegs: represents a set of registers that can be used to allocate the
+// specified GenTree in any point in time.
```

Comment:
I don't know what "in any point in time" means. Also, it is a bit misleading to me because it seems to imply that it is taking some specific requirements of the GenTree instance into account, when in fact it is only taking into account the type of the node (generalized to include the sub-types of a multi-reg call node). It might be more accurate to say "returns the set of registers that can accommodate the type of the given node." And then add a "Note:" that says something like "If the node is a multi-reg call node, the full set of registers must be determined by looking at types of the individual return registers." Because this can be somewhat confusing and complex, it might be worth it to further add something like "In that case, the registers may include registers from different register sets, and will not be limited to the actual ABI return registers."